README.md

```diff
@@ -37,6 +37,64 @@ go get github.com/aws-controllers-k8s/dev-tools/cmd/ackdev
 
 Call `ackdev help` for detailed usage instructions.
 
+### Setup
+
+To be able to use `ackdev` you'll have to run `ackdev setup` before any other command.
+
+For example
+```bash
+ackdev setup --root-directory $WORKDIR --services s3,ecr,sqs,sns
+```
+
+The setup command will simply create a yaml file named `$HOME/.ackdev.yaml`
+(you can choose a different file path `--config-file`)
+
+**NOTE**: If you are a contributor you probably want to leave `--root-directory` empty which will default to `$GOPATH/src/github.com/aws-controllers-k8s`
+(make sure that this directory exists).
+If you only want to test the tool, 
+We recommended you to set your `$WORKDIR` to a different directory.
+
+> Will my projects work/compile if I use a different directory than `$GOPATH/src/github.com/aws-controllers-k8s` ?
+
+Yes, even if it's not inside `GOPATH`, it will work as expected.
+Since Go 1.11.4 the Go compiler doesn't rely **entirely** on GOPATH to build projects.
+
+
+The generated configuration file will look like:
+
+``` yaml
+rootDirectory: /home/amine/go/source/github.com/aws-controllers-k8s/dev-tools
+git:
+  sshKeyPath: ""
```

Comment:
is this `sshKeyPath` really required here? I mean, shouldn't we keep Git-specific connectivity/configuration information in the typical `$HOME/.gitconfig` files and have our Git-executing libraries just assume Git configuration values from that file?

I believe the default behaviour if using `git` over SSH is to use private keys that are in the `$HOME/.ssh/` directory.

Reply:
> is this sshKeyPath really required here? 

If we are using pure [go-git](github.com/go-git/go-git), yes. Calling `go-git.Clone(...)` doesn't look for the values stored in `$HOME/.gitconfig`. ( i think that this is more specific to the `git` CLI)
Currently `sshKeyPath` is used to [create a `ssh.Signer`](https://github.com/A-Hilaly/ack-dev-tools/blob/repository-experiments/pkg/util/pem.go#L14-L56) - which will be passed later passed to `go-git.Clone` options.

Reply:
One alternative solution that will work for sure, is to execute the `git clone` command instead instead of using `go-git.Clone(...)`

Reply:
@jaypipes I will change the `Cloner` implementation to use `git clone` instead of "gogit" - this way we can remove the `sshKeyPath` from the configuration